.github/workflows/soak-testing.yml

```diff
@@ -0,0 +1,233 @@
+# Pre-requisites:
+# - AWS Account with correct permissions
```

Comment:
The permissions look like this for now, but they could most likely be more strict.

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": "logs:PutLogEvents",
            "Resource": "arn:aws:logs:*:<ACCOUNT_ID>:log-group:*:log-stream:*"
        },
        {
            "Sid": "VisualEditor1",
            "Effect": "Allow",
            "Action": "logs:CreateLogStream",
            "Resource": "arn:aws:logs:*:<ACCOUNT_ID>:log-group:*"
        },
        {
            "Sid": "VisualEditor2",
            "Effect": "Allow",
            "Action": [
                "cloudwatch:PutMetricAlarm",
                "cloudwatch:DeleteAlarms",
                "cloudwatch:GetMetricData",
                "cloudwatch:DescribeAlarms",
                "sts:*",
                "cloudwatch:GetMetricWidgetImage"
            ],
            "Resource": "*"
        }
    ]
}
```

Reply:
I guess we need to add here

https://github.com/aws-observability/aws-otel-ops/blob/main/python.tf

Can you figure out the minimal permissions?

For reference, we have many custom permissions for aws-otel-lambda repo

https://github.com/aws-observability/aws-otel-ops/blob/main/lambda.tf#L29

Reply:
> I guess we need to add here

I'll make sure I take the time to understand what that file is doing, I'm fairly unfamiliar with terraform but it seems worthwhile since this is the pattern we are following.

> Can you figure out the minimal permissions?

Some of them have to be `*` because things like the `log-stream-name` are dynamic. Others like the `log-group` are repository specific for now, but I think I'll make it more general so we can apply it to other language-sdk repos more easily.

> For reference, we have many custom permissions for aws-otel-lambda repo

Thanks! I'll look into this, especially for the "role" which I was having trouble figuring out earlier.

Reply:
I updated the permissions we need in my original comment. Added a note about it being permissions for just the Python repo or if we want we can make the policy general for all language repos.